doc/tutorials/your_first_project/your_first_project.rst

```diff
@@ -0,0 +1,275 @@
+
+After that we have the normal call to initialize rclcpp and then we create our Node.
+
+.. code-block:: C++
+
+  auto const node = std::make_shared<rclcpp::Node>(
+    "hello_moveit",
+    rclcpp::NodeOptions().automatically_declare_parameters_from_overrides(true)
+  );
+
+The first argument is the string that ROS will use to make a unique node.
+The second is needed for MoveIt because of how we use ROS Parameters.
+
+Before we can start use our ROS Node we have to give it to a spinning executor.
+This is what will enable ROS to call callbacks to update our Robot State among other things.
+
+.. code-block:: C++
+
+  auto thread = std::thread([node]() {
+    rclcpp::executors::SingleThreadedExecutor executor;
+    executor.add_node(node);
+    executor.spin();
+  });
+
+Lastly we have the code to shutdown ROS and cleanup our thread.
+
+3 Plan and Execute using MoveGroupInterface
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+After the creation of the thread add this block of code:
+
+.. code-block:: C++
+
+  // Create the MoveIt MoveGroup Interface
+  using moveit::planning_interface::MoveGroupInterface;
+  auto move_group_interface = MoveGroupInterface(node, "panda_arm");
+
+  // Set a target Pose
+  auto const target_pose = []{
+    geometry_msgs::msg::Pose msg;
+    msg.orientation.w = 1.0;
+    msg.position.x = 0.28;
+    msg.position.y = -0.2;
+    msg.position.z = 0.5;
+    return msg;
+  }();
+  move_group_interface.setPoseTarget(target_pose);
+
+  // Create a plan to that target pose
+  auto const [success, plan] = [&move_group_interface]{
+    moveit::planning_interface::MoveGroupInterface::Plan msg;
+    auto const ok = static_cast<bool>(move_group_interface.plan(msg));
+    return std::make_pair(ok, msg);
+  }();
+
+  // Execute the plan
+  if(success) {
+    move_group_interface.execute(plan);
+  } else {
+    RCLCPP_ERROR(logger, "Planing failed!");
+  }
+
+3.1 Build and Run
+~~~~~~~~~~~~~~~~~
+
+Just like before we need to build the code before we can run it.
+
+In the workspace directory ``ws_moveit2`` run this command:
+
+.. code-block:: bash
+
+  colcon build --mixin debug
+
+After this succeeds, we need to re-use the demo launch file from the previous tutorial to start RViz and the MoveGroup node.
+In a separate terminal, source the workspace and then execute this:
+
+.. code-block:: bash
+
+  ros2 launch moveit2_tutorials demo.launch.py
+
+Then in the ``Displays`` window under ``MotionPlanning/Planning Request`` uncheck the box ``Query Goal State``.
+
+.. image:: rviz_1.png
+   :width: 300px
+
+In a third terminal source the workspace and run your program.
+
+.. code-block:: bash
+
+  ros2 run hello_moveit hello_moveit
+
+This should cause the robot in RViz to move and end up in this pose:
+
+.. image:: rviz_2.png
+   :width: 300px
+
+3.2 Examine the code
+~~~~~~~~~~~~~~~~~~~~
+
+The first thing we do is create the MoveGroupInterface.
```

Comment:
```suggestion
The first thing we do is create the MoveGroupInterface. This object will be used to interact with move_group, which allow us to plan and execute trajectories.
```

Maybe a bit more explanation would be nice? Otherwise it reads a bit mysterious. Can even link to other docs if applicable.